packages/TimePicker/src/TimePicker.js

```diff
@@ -1,130 +1,128 @@
-    labelMinutes: PropTypes.string,
-    labelPeriod: PropTypes.string,
-    labelPM: PropTypes.string,
-    onChange: PropTypes.func,
-    prefix: PropTypes.string,
-  };
-
-  static defaultProps = {
-    ariaLabel: "Time (hh:mm)",
-    defaultIsVisible: false,
-    defaultValue: null,
-    labelCustom: "Custom",
-    labelError: "Invalid Format",
-    labelHours: "Hours",
-    labelMinutes: "Minutes",
-    labelPeriod: "Period",
-    labelAM: "am",
-    labelPM: "pm",
-    isDisabled: false,
-    onChange: () => {},
-    prefix: "timePicker",
-  };
-
-  constructor(props) {
-    super(props);
-    this.name = `${props.prefix}${btoa(Math.random()).substring(0, 8)}`;
-    this.state = {
+const propTypes = {
+  /** Descriptive a11y text for assistive technologies. */
+  a11yText: PropTypes.string,
+
+  /** If the TimePicker is set to visible. */
+  defaultIsOpen: PropTypes.bool,
+
+  /** Sets the default value for the TimePicker */
+  defaultValue: PropTypes.string,
+
+  /** If the TimePicker is disabled. */
+  isDisabled: PropTypes.bool,
+
+  /** Callback to be executed when the value is changed. */
+  onChange: PropTypes.func,
+
+  /** Callback to be executed when there is an error. */
+  onError: PropTypes.func,
+
+  /** */
+  prefix: PropTypes.string,
+};
+
+const defaultProps = {
+  a11yText: "Time (hh:mm)",
+  defaultIsOpen: false,
+  defaultValue: null,
+  isDisabled: false,
+  onChange: () => {},
+  onError: () => {},
+  prefix: "timePicker",
+};
+
+function TimePicker(props) {
+  const { a11yText, defaultIsOpen, defaultValue, isDisabled, onChange, onError, prefix, ...moreProps } = props;
+
+  const [isOpen, setIsOpen] = React.useState(defaultIsOpen);
+  const [time, setTime] = React.useState(() => {
+    if (defaultValue) {
+      const defaultTime = getExplodeTime(defaultValue);
+      return {
+        hh: defaultTime.hh,
+        mm: defaultTime.mm,
+        period: defaultTime.period,
+      };
+    }
+    return {
       hh: null,
-      isPristine: true,
-      isVisible: this.props.defaultIsVisible,
       mm: null,
       period: null,
-      timeStr: null,
-      isTabIndexActive: 0,
-      value: "",
     };
-  }
-
-  static getDerivedStateFromProps(props, state) {
-    if (props.defaultValue && !state.value && state.isPristine) {
-      const time = getExplodeTime(props.defaultValue);
-
-      if (time.error) {
-        return {
-          ...time,
-          error: props.labelError,
-          isVisible: false,
-          isPristine: false,
-        };
+  });
+  const [timeStr, setTimeStr] = React.useState(null);
+  const [value, setValue] = React.useState(defaultValue);
+
+  React.useEffect(() => {
+    if (defaultValue) {
+      const { hh, mm, period, error, timeStr: newTimeStr, value: newValue } = getExplodeTime(defaultValue);
+
+      if (error) {
+        setIsOpen(false);
+        // TODO: not required if we assume consumer always pass in valid value
+        // onError(error);
+        console.error();
```

Comment:
maybe we need a short description here like `TimePicker - invalid defaultValue, please check your defaultValue passed in`

Reply:
Also the TODO and onError can be removed since we console out the error